data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -108,3 +110,174 @@ def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
+                    != constants.CEJST_SCORE_COLUMN_NAME
+                )
+                codebook_dictionary[
+                    field_information.new_label_in_codebook
+                ].append(np.nan)
+    return pd.DataFrame(codebook_dictionary).set_index(
+        constants.CEJST_SCORE_COLUMN_NAME
+    )
+
+
+def _get_datatype(
+    input_column_name: str,
+    input_column_type: str,
+    percentile_string: str = field_names.PERCENTILE_FIELD_SUFFIX,
+    loss_rate_string: str = constants.LOSS_RATE_STRING,
+) -> str:
+    """Helper to convert datatype"""
+    return_column_type = input_column_type
+    if percentile_string in input_column_name:
+        return_column_type = "percentile"
+    elif loss_rate_string in input_column_name:
+        return_column_type = "rate"
+    return return_column_type
+
+
+def _get_calculation_notes(column_name: str) -> str:
+    """Produces calculation notes
+
+    Note: eventually, this will either be programmatically set, or will be included in the yaml, depending on
+    the refactor that we do
+    """
+    calculation_notes = []
+    if field_names.PERCENTILE_FIELD_SUFFIX in column_name:
+        calculation_notes += [constants.PERCENTILE_EXPLANATION]
+    if constants.LOW_STRING in column_name:
+        calculation_notes += [constants.LOW_PERCENTILE_EXPLANATION]
+    if constants.ISLAND_STRING in column_name:
+        calculation_notes += [constants.ISLAND_AREAS_EXPLANATION]
+    return " ".join(calculation_notes)
+
+
+def create_codebook(
+    downloadable_csv_config: dict, excel_config: dict, extra_score_info: dict
+) -> pd.DataFrame:
+    """Runs through all logic of creating the codebook.
+
+    First it reads in each component yaml file for the codebook.
+    Then it merges all of them.
+    Finally, it applies any transforms to the columns (like getting the
+        datatype or adding calculation_notes.
+    """
+    CodebookLabelFields = namedtuple(
+        "CodebookLabelFields",
+        ["new_label_in_codebook", "existing_yaml_label"],
+    )
+
+    # parse data from component yamls
+    csv_codes_df = _create_df_from_yaml_contents(
+        fields_list_from_yaml=downloadable_csv_config,
+        fields_to_store_in_codebook=[
+            CodebookLabelFields(
+                new_label_in_codebook=constants.CSV_LABEL_FIELD,
+                existing_yaml_label="label",
+            ),
+            CodebookLabelFields(
+                new_label_in_codebook=constants.CSV_FORMAT,
+                existing_yaml_label="format",
+            ),
+        ],
+    )
+
+    excel_codes_df = _create_df_from_yaml_contents(
+        fields_list_from_yaml=excel_config,
+        fields_to_store_in_codebook=[
+            CodebookLabelFields(
+                new_label_in_codebook=constants.EXCEL_LABEL_FIELD,
+                existing_yaml_label="label",
+            )
+        ],
+    )
+
+    score_detail_notes_df = _create_df_from_yaml_contents(
```

Comment:
on the hard problem of naming things, consistency wherever possible is essential. you've referred to the information in `extra_score_info` by three different names: `score_fields.yml`, `extra_score_info`, and `score_detail_notes_df`. 

you can instead have consistency like `field_descriptions_for_codebook.yml`, `field_descriptions_for_codebook`, and `field_descriptions_for_codebook_df` that will make it much more obvious what's happening

Reply:
🎉 sounds good and makes sense!